src/lib.rs

```diff
@@ -1608,6 +1608,53 @@ pub trait Itertools : Iterator {
         None
     }
 
+    /// Check whether the iterator contains an item.
+    ///
+    /// If the iterator contains the item prior to its end,
+    /// this method will short-circuit and only partially
+    /// exhaust the iterator.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![
+    ///      "foo".to_owned(),
+    ///      "bar".to_owned(),
+    ///      "baz".to_owned(),
+    /// ];
+    /// // this could get expensive:
+    /// assert!(data.contains(&"foo".to_owned()));
+    /// // this, less so:
+    /// assert!(data.iter().contains("foo"));
+    ///
+    /// // now the not-as-motivating tests involving Copy data:
+    ///
+    /// let data = vec![4usize, 5, 1, 3, 0, 2];
+    /// assert!(data.iter().contains(&4));
+    /// assert!(!data.iter().contains(&6));
+    /// for x in (0..50) {
+    ///     assert_eq!(data.contains(&x), data.iter().contains(&x));
+    /// }
+    ///
+    /// let mut it = data.iter();
+    /// assert!(!it.contains(&6));
+    /// assert_eq!(it.next(), None);
+    ///
+    /// let mut it = data.iter();
+    /// assert!(it.contains(&3));
+    /// assert_eq!(it.next(), Some(&0));
+    ///
+    /// let data : Option<usize> = None;
+    /// assert!(!data.into_iter().contains(0));
+    /// ```
+    fn contains<Q>(&mut self, query: Q) -> bool
+    where
+        Self: Sized,
+        Self::Item: PartialEq<Q>,
+    {
+        self.any(|x| x == query)
+    }
```

Comment:
Concretely, @phimuemue is proposing this, instead ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7a59790f362e1517873667f4a245679b)):
```suggestion
    /// ```
    /// #[derive(PartialEq, Debug)]
    /// enum Enum { A, B, C, D, E, }
    /// 
    /// let mut iter = vec![Enum::A, Enum::B, Enum::C, Enum::D].into_iter();
    /// 
    /// // search `iter` for `B`
    /// assert_eq!(iter.contains(Enum::B), true);
    /// // `B` was found, so the iterator now rests at the item after `B` (i.e, `C`).
    /// assert_eq!(iter.next(), Some(Enum::C));
    /// 
    /// // search `iter` for `E`
    /// assert_eq!(iter.contains(Enum::E), false);
    /// // `E` wasn't found, so `iter` is now exhausted
    /// assert_eq!(iter.next(), None);
    /// ```
    fn contains<Q>(&mut self, query: &Q) -> bool
    where
        Self: Sized,
        Self::Item: Borrow<Q>,
        Q: PartialEq,
    {
        self.any(|x| x.borrow() == query)
    }
```

This seems like a sensible improvement to me!